src/components/Text/Text.tsx

```diff
@@ -1,37 +1,80 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React from 'react';
 import styles from './Text.module.css';
 
-export interface Props {
-  /**
-   * The rendered tag name of the Heading. The tag name should always relate to the [document outline](http://html5doctor.com/outlines/) and a tag name should never be chosen for its default aesthetic qualities. If a specific style is desired, use the `size` prop to manipulate the Heading style.
-   */
-  as: 'p' | 'span';
-  /**
-   * The child node(s)
-   */
-  children?: ReactNode;
+export type Size =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'body'
+  | 'sm'
+  | 'xs'
+  | 'caption'
+  | 'overline';
+
+export type Color =
+  | 'alert'
+  | 'base'
+  | 'brand'
+  | 'info'
+  | 'inherit'
+  | 'neutral'
+  | 'success'
+  | 'warning'
+  | 'white';
+
+export type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * Controls whether to render text inline (defaults to "p");
    */
+  as?: 'p' | 'span';
+  children: React.ReactNode;
   className?: string;
-}
+  color?: Color;
+  size?: Size;
+  spacing?: 'none' | 'half' | '1x' | '2x';
```

Comment:
could we remove this prop? not so useful with TW utility classes now

Reply:
Yup, sounds good

Reply:
One nice thing about using props though is we're specifying which levels are appropriate for the component. We can't stop you from using 12px but we encourage you to pick from this specific set. When we were migrating over the headings in traject, I found it useful to have a short list to pick from.